shapely/tests/geometry/test_decimal.py

```diff
@@ -0,0 +1,84 @@
+from decimal import Decimal
+
+import pytest
+
+from shapely import (
+    GeometryCollection,
+    LinearRing,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Point,
+    Polygon,
+)
+
+
+def to_decimal(x):
+    if isinstance(x, (int, float)):
+        return Decimal(x)
+    if isinstance(x, tuple):
+        return tuple([to_decimal(x) for x in x])
+    if isinstance(x, list):
+        return list([to_decimal(x) for x in x])
+    return x
+
+
+def test_to_decimal():
+    assert to_decimal(1) == Decimal(1)
+    assert to_decimal([1]) == [Decimal(1)]
+    assert to_decimal([1, 2]) == [Decimal(1), Decimal(2)]
+    assert to_decimal((1, 2)) == (Decimal(1), Decimal(2))
+    assert to_decimal([(1, 2), [3], 4, [5, 6, 7]]) == [
+        (Decimal(1), Decimal(2)),
+        [Decimal(3)],
+        Decimal(4),
+        [Decimal(5), Decimal(6), Decimal(7)],
+    ]
+
+
+items2d = [
+    [(0.0, 0.0), (70.0, 120.0), (140.0, 0.0), (0.0, 0.0)],
+    [(60.0, 80.0), (80.0, 80.0), (70.0, 60.0), (60.0, 80.0)],
+    [(30.0, 10.0), (50.0, 10.0), (40.0, 30.0), (30.0, 10.0)],
+    [(90.0, 10), (110.0, 10.0), (100.0, 30.0), (90.0, 10.0)],
+]
+
+items3d = [
+    [(0.0, 0.0, 1.0), (70.0, 120.0, 2.0), (140.0, 0.0, 3.0), (0.0, 0.0, 1.0)],
+    [(60.0, 80.0, 1.0), (80.0, 80.0, 2.0), (70.0, 60.0, 3.0), (60.0, 80.0, 1.0)],
+    [(30.0, 10.0, 2.0), (50.0, 10.0, 3.0), (40.0, 30.0, 4.0), (30.0, 10.0, 2.0)],
+    [(90.0, 10, 3.0), (110.0, 10.0, 4.0), (100.0, 30.0, 5.0), (90.0, 10.0, 6.0)],
+]
+
+
+@pytest.mark.parametrize("items", [items2d, items3d])
+def test_decimal(items):
+    assert Point(items[0][0]) == Point(to_decimal(items[0][0]))
```

Comment:
I don't agree with the use of to_decimal here. It's unnecessarily indirect. Why not just give instances of Decimal, explicitly?

Reply:
I'm not sure if you were referring to the use of the function `to_decimal` in general in this test or just to the `Point` test.
Anyway, I've parametrized the test as you requested above. 
If you'd prefer to remove `to_decimal` I could hardcode `Decimal` in copies of `items2d` and `items3d` and remove `to_decimal` - otherwise please explain what you meant please.

Reply:
Here's what I've come to understand about using convenience functions in tests. I used to do this somewhat often. But then a year or two later when I wanted to refactor code or update a feature, I would discover that the convenience function wasn't appropriate anymore and actually hindered the refactoring work. This is the awful truth about tests: they can easily make future changes hard. Shapely needs tests, but not too many, and not too fancy.

Please do remove to_decimal.

Reply:
This is important work @idanmiara and you've solved the problem well! I want to help you get the tests right.

Reply:
@sgillies thanks, I see your point. while removing `to_decimal` I've added another test for mixed `decimal` and non-decimal inputs.